src/components/pages/layouts/MainLayout.tsx

```diff
@@ -6,30 +6,38 @@ import Copyright from '../../shared/Copyright'
 import { Link, Outlet } from 'react-router-dom'
 import NavBarLinkSection from '../../shared/navbar/NavBarLinkSection'
 import ThemeSwitcher from '../../shared/navbar/ThemeSwitcher'
+import useOnlineStatus from '../../../hooks/use-online-status'
+import Banner from '../../shared/Banner'
+import OfflineNotification from '../../shared/OfflineNotification'
 
-const MainLayout = () => (
-  <>
-    <AppBar position="static">
-      <Toolbar>
-        <IconButton
-          size="large"
-          edge="start"
-          sx={{ mr: 2 }}
-          component={Link}
-          to='/'
-        >
-          <LandscapeIcon/>
-        </IconButton>
-        <NavBarLinkSection/>
-        <NavBarUserSection/>
-        <ThemeSwitcher/>
-      </Toolbar>
-    </AppBar>
+const MainLayout = () => {
+  const isOnline = useOnlineStatus()
 
-    <Outlet/>
+  const offlineBannerContent = isOnline ? null : <OfflineNotification/>
 
-    <Copyright/>
-  </>
-)
+  return (<>
+      <AppBar position="static">
+        <Toolbar>
+          <IconButton
+            size="large"
+            edge="start"
+            sx={{ mr: 2 }}
+            component={Link}
+            to='/'
+          >
+            <LandscapeIcon/>
+          </IconButton>
+          <NavBarLinkSection/>
+          <NavBarUserSection/>
+          <ThemeSwitcher/>
+        </Toolbar>
+      </AppBar>
+      <Banner bannerContent={offlineBannerContent}/>
```

Comment:
@Saela But then we'd always render the `<Banner>` element (even though it does not return any content if it is null). Is there no way that we could just render the whole `<Banner>` element conditionally instead of just its childprop?

Reply:
@Tugark Resolved in pr #94